### ISSUE TITLE
Can't install contourpy 1.1.0

### ISSUE THREAD

**Issue by andronat** [nix-community/poetry2nix#1192]
### Describe the issue

I'm getting this error when I'm trying to install `contourpy 1.1.0`:

```
Creating a wheel...
[33mWARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.[0m[33m
[0m[31mERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.[0m[31m
[0m
```

### Additional context

In my `flake.nix` I have the following, where `pkgs` point to `nixos-22.11` and  `newPkgs` to `nixos-23.05` as `meson 1.1.0` is required:
```nix
  contourpy = super.contourpy.overrideAttrs (old: {
    nativeBuildInputs = (old.nativeBuildInputs or [ ]) ++ [
      newPkgs.meson
      pkgs.python310.pkgs.pybind11
      pkgs.python310.pkgs.meson-python
      pkgs.pkg-config
      pkgs.cmake
    ];
  });
```

`poetry.lock`:
```toml
[[package]]
name = "contourpy"
version = "1.1.0"
description = "Python library for calculating contours of 2D quadrilateral grids"
category = "main"
optional = false
python-versions = ">=3.8"
files = [
    {file = "contourpy-1.1.0-cp310-cp310-macosx_10_9_x86_64.whl", hash = "sha256:89f06eff3ce2f4b3eb24c1055a26981bffe4e7264acd86f15b97e40530b794bc"},
    {file = "contourpy-1.1.0-cp310-cp310-macosx_11_0_arm64.whl", hash = "sha256:dffcc2ddec1782dd2f2ce1ef16f070861af4fb78c69862ce0aab801495dda6a3"},
    {file = "contourpy-1.1.0-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:25ae46595e22f93592d39a7eac3d638cda552c3e1160255258b695f7b58e5655"},
    {file = "contourpy-1.1.0-cp310-cp310-manylinux_2_17_ppc64le.manylinux2014_ppc64le.whl", hash = "sha256:17cfaf5ec9862bc93af1ec1f302457371c34e688fbd381f4035a06cd47324f48"},
    {file = "contourpy-1.1.0-cp310-cp310-manylinux_2_17_s390x.manylinux2014_s390x.whl", hash = "sha256:18a64814ae7bce73925131381603fff0116e2df25230dfc80d6d690aa6e20b37"},
    {file = "contourpy-1.1.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:90c81f22b4f572f8a2110b0b741bb64e5a6427e0a198b2cdc1fbaf85f352a3aa"},
    {file = "contourpy-1.1.0-cp310-cp310-musllinux_1_1_x86_64.whl", hash = "sha256:53cc3a40635abedbec7f1bde60f8c189c49e84ac180c665f2cd7c162cc454baa"},
    {file = "contourpy-1.1.0-cp310-cp310-win_amd64.whl", hash = "sha256:1f795597073b09d631782e7245016a4323cf1cf0b4e06eef7ea6627e06a37ff2"},
    {file = "contourpy-1.1.0-cp311-cp311-macosx_10_9_x86_64.whl", hash = "sha256:0b7b04ed0961647691cfe5d82115dd072af7ce8846d31a5fac6c142dcce8b882"},
    {file = "contourpy-1.1.0-cp311-cp311-macosx_11_0_arm64.whl", hash = "sha256:27bc79200c742f9746d7dd51a734ee326a292d77e7d94c8af6e08d1e6c15d545"},
    {file = "contourpy-1.1.0-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:052cc634bf903c604ef1a00a5aa093c54f81a2612faedaa43295809ffdde885e"},
    {file = "contourpy-1.1.0-cp311-cp311-manylinux_2_17_ppc64le.manylinux2014_ppc64le.whl", hash = "sha256:9382a1c0bc46230fb881c36229bfa23d8c303b889b788b939365578d762b5c18"},
    {file = "contourpy-1.1.0-cp311-cp311-manylinux_2_17_s390x.manylinux2014_s390x.whl", hash = "sha256:e5cec36c5090e75a9ac9dbd0ff4a8cf7cecd60f1b6dc23a374c7d980a1cd710e"},
    {file = "contourpy-1.1.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:1f0cbd657e9bde94cd0e33aa7df94fb73c1ab7799378d3b3f902eb8eb2e04a3a"},
    {file = "contourpy-1.1.0-cp311-cp311-musllinux_1_1_x86_64.whl", hash = "sha256:181cbace49874f4358e2929aaf7ba84006acb76694102e88dd15af861996c16e"},
    {file = "contourpy-1.1.0-cp311-cp311-win_amd64.whl", hash = "sha256:fb3b7d9e6243bfa1efb93ccfe64ec610d85cfe5aec2c25f97fbbd2e58b531256"},
    {file = "contourpy-1.1.0-cp38-cp38-macosx_10_9_x86_64.whl", hash = "sha256:bcb41692aa09aeb19c7c213411854402f29f6613845ad2453d30bf421fe68fed"},
    {file = "contourpy-1.1.0-cp38-cp38-macosx_11_0_arm64.whl", hash = "sha256:5d123a5bc63cd34c27ff9c7ac1cd978909e9c71da12e05be0231c608048bb2ae"},
    {file = "contourpy-1.1.0-cp38-cp38-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:62013a2cf68abc80dadfd2307299bfa8f5aa0dcaec5b2954caeb5fa094171103"},
    {file = "contourpy-1.1.0-cp38-cp38-manylinux_2_17_ppc64le.manylinux2014_ppc64le.whl", hash = "sha256:0b6616375d7de55797d7a66ee7d087efe27f03d336c27cf1f32c02b8c1a5ac70"},
    {file = "contourpy-1.1.0-cp38-cp38-manylinux_2_17_s390x.manylinux2014_s390x.whl", hash = "sha256:317267d915490d1e84577924bd61ba71bf8681a30e0d6c545f577363157e5e94"},
    {file = "contourpy-1.1.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:d551f3a442655f3dcc1285723f9acd646ca5858834efeab4598d706206b09c9f"},
    {file = "contourpy-1.1.0-cp38-cp38-musllinux_1_1_x86_64.whl", hash = "sha256:e7a117ce7df5a938fe035cad481b0189049e8d92433b4b33aa7fc609344aafa1"},
    {file = "contourpy-1.1.0-cp38-cp38-win_amd64.whl", hash = "sha256:d4f26b25b4f86087e7d75e63212756c38546e70f2a92d2be44f80114826e1cd4"},
    {file = "contourpy-1.1.0-cp39-cp39-macosx_10_9_x86_64.whl", hash = "sha256:bc00bb4225d57bff7ebb634646c0ee2a1298402ec10a5fe7af79df9a51c1bfd9"},
    {file = "contourpy-1.1.0-cp39-cp39-macosx_11_0_arm64.whl", hash = "sha256:189ceb1525eb0655ab8487a9a9c41f42a73ba52d6789754788d1883fb06b2d8a"},
    {file = "contourpy-1.1.0-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:9f2931ed4741f98f74b410b16e5213f71dcccee67518970c42f64153ea9313b9"},
    {file = "contourpy-1.1.0-cp39-cp39-manylinux_2_17_ppc64le.manylinux2014_ppc64le.whl", hash = "sha256:30f511c05fab7f12e0b1b7730ebdc2ec8deedcfb505bc27eb570ff47c51a8f15"},
    {file = "contourpy-1.1.0-cp39-cp39-manylinux_2_17_s390x.manylinux2014_s390x.whl", hash = "sha256:143dde50520a9f90e4a2703f367cf8ec96a73042b72e68fcd184e1279962eb6f"},
    {file = "contourpy-1.1.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:e94bef2580e25b5fdb183bf98a2faa2adc5b638736b2c0a4da98691da641316a"},
    {file = "contourpy-1.1.0-cp39-cp39-musllinux_1_1_x86_64.whl", hash = "sha256:ed614aea8462735e7d70141374bd7650afd1c3f3cb0c2dbbcbe44e14331bf002"},
    {file = "contourpy-1.1.0-cp39-cp39-win_amd64.whl", hash = "sha256:438ba416d02f82b692e371858143970ed2eb6337d9cdbbede0d8ad9f3d7dd17d"},
    {file = "contourpy-1.1.0-pp38-pypy38_pp73-macosx_10_9_x86_64.whl", hash = "sha256:a698c6a7a432789e587168573a864a7ea374c6be8d4f31f9d87c001d5a843493"},
    {file = "contourpy-1.1.0-pp38-pypy38_pp73-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:397b0ac8a12880412da3551a8cb5a187d3298a72802b45a3bd1805e204ad8439"},
    {file = "contourpy-1.1.0-pp38-pypy38_pp73-win_amd64.whl", hash = "sha256:a67259c2b493b00e5a4d0f7bfae51fb4b3371395e47d079a4446e9b0f4d70e76"},
    {file = "contourpy-1.1.0-pp39-pypy39_pp73-macosx_10_9_x86_64.whl", hash = "sha256:2b836d22bd2c7bb2700348e4521b25e077255ebb6ab68e351ab5aa91ca27e027"},
    {file = "contourpy-1.1.0-pp39-pypy39_pp73-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:084eaa568400cfaf7179b847ac871582199b1b44d5699198e9602ecbbb5f6104"},
    {file = "contourpy-1.1.0-pp39-pypy39_pp73-win_amd64.whl", hash = "sha256:911ff4fd53e26b019f898f32db0d4956c9d227d51338fb3b03ec72ff0084ee5f"},
    {file = "contourpy-1.1.0.tar.gz", hash = "sha256:e53046c3863828d21d531cc3b53786e6580eb1ba02477e8681009b6aa0870b21"},
]

[package.dependencies]
numpy = ">=1.16"

[package.extras]
bokeh = ["bokeh", "selenium"]
docs = ["furo", "sphinx-copybutton"]
mypy = ["contourpy[bokeh,docs]", "docutils-stubs", "mypy (==1.2.0)", "types-Pillow"]
test = ["Pillow", "contourpy[test-no-images]", "matplotlib"]
test-no-images = ["pytest", "pytest-cov", "wurlitzer"]

```

Full log:
```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing pip-build-hook
Using pipBuildPhase
Using pipShellHook
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/4ir661y1xw9svbxb91xfnll8ibjbdna3-contourpy-1.1.0.tar.gz
source root is contourpy-1.1.0
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file contourpy-1.1.0/tests/util_test.py
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Removing path dependencies
Finished removing path dependencies
Removing git dependencies
Finished removing git dependencies
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
meson flags: --buildtype=plain         --libdir=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/lib --libexecdir=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/libexec         --bindir=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/bin --sbindir=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/sbin         --includedir=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/include         --mandir=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/share/man --infodir=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/share/info         --localedir=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/share/locale         -Dauto_features=enabled         -Dwrap_mode=nodownload         --prefix=/nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0  
[1mThe Meson build system[0m
Version: 1.1.0
Source dir: [1m/tmp/nix-build-python3.10-contourpy-1.1.0.drv-0/contourpy-1.1.0[0m
Build dir: [1m/tmp/nix-build-python3.10-contourpy-1.1.0.drv-0/contourpy-1.1.0/build[0m
Build type: [1mnative build[0m
Project name: [1mcontourpy[0m
Project version: [1m1.1.0[0m
C++ compiler for the host machine: [1mg++[0m (gcc 9.5.0 "g++ (GCC) 9.5.0")
C++ linker for the host machine: [1mg++[0m ld.bfd 2.39
Host machine cpu family: [1maarch64[0m
Host machine cpu: [1maarch64[0m
Program python3 found: [1;32mYES[0m (/nix/store/kvc8g2i596glj3f7m3mwhm7shq4sd02s-python3-3.10.11/bin/python3.10)
Found pkg-config: [1m/nix/store/0npqcr368vkcg57llkxg7d4q0vj6ywbq-pkg-config-wrapper-0.29.2/bin/pkg-config[0m (0.29.2)
[1mpybind11-config[0m found: [1;31mNO[0m
Found CMake: [1m/nix/store/4p8rlpmyigq6lzcw6w9vcx1dkjszah5g-cmake-3.24.3/bin/cmake[0m (3.24.3)
Run-time dependency [1mpybind11[0m found: [1;32mYES[0m [36m2.10.0[0m
Configuring [1m_build_config.py[0m using configuration
Build targets in project: [1m1[0m

contourpy [36m1.1.0[0m

  [1mBuild config[0m
    host_cpu     : aarch64
    build_cpu    : aarch64
    cross_build  : False
    optimization : plain

  [1mUser defined options[0m
    auto_features: enabled
    bindir       : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/bin
    buildtype    : plain
    includedir   : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/include
    infodir      : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/share/info
    libdir       : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/lib
    libexecdir   : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/libexec
    localedir    : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/share/locale
    mandir       : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/share/man
    prefix       : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0
    sbindir      : /nix/store/ismbv29jkd275vsdzid5zzd92wp1646h-python3.10-contourpy-1.1.0/sbin
    wrap_mode    : nodownload

Found ninja-1.11.1 at /nix/store/h6j9b4ydmdfanrk5h24m3zn19vf34pvi-ninja-1.11.1/bin/ninja
meson: enabled parallel building
meson: enabled parallel installing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing pipBuildPhase
Creating a wheel...
[33mWARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.[0m[33m
[0m[31mERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.[0m[31m
[0m
```

**Comment by l0b0** [comment]
Would using a [wheel](https://github.com/nix-community/poetry2nix#faq) work for you?

**Comment by AlexChalk** [comment]
I'm not an expert (on meson or poetry2nix), but I think the place you'd want to PR is at nixpkgs: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/python-modules/contourpy/default.nix. 

Contourpy only recently updated their build system, and the nix build specification hasn't caught up yet (it is still for 1.0.7).

If you wanted to have a go at this yourself, these docs look like they might be useful: https://github.com/NixOS/nixpkgs/blob/a5ca6a0f2ee8b2c8aa2cca64a06bd5652ad28a4b/doc/hooks/meson.section.md.

**Comment by jzbor** [comment]
> Here's an example of a nix specification for a python package built with meson: https://github.com/NixOS/nixpkgs/blob/a5ca6a0f2ee8b2c8aa2cca64a06bd5652ad28a4b/pkgs/development/python-modules/pycairo/default.nix

Funny enough I think I have run into this exact issue with pycairo:
```
       last 10 log lines:
       >
       > Found ninja-1.11.1 at /nix/store/3vdyiqq685f3z179i4jfc4pxi50dizy3-ninja-1.11.1/bin/ninja
       > meson: enabled parallel building
       > meson: enabled parallel installing
       > building
       > Executing pipBuildPhase
       > Creating a wheel...
       > WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
       > ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.
       > 
       For full logs, run 'nix log /nix/store/hp2abiz1lry77rhjfbl227cgl0i9s38f-python3.10-pycairo-1.24.0.drv'.
```

**Comment by vleni** [comment]
I'm seeing a similar error
```
Processing /build/contourpy-1.1.1
  Running command Preparing metadata (pyproject.toml)
  The Meson build system
  Version: 0.64.1
  Source dir: /build/contourpy-1.1.1
  Build dir: /build/contourpy-1.1.1/.mesonpy-6x3o8xj_/build
  Build type: native build

  ../../meson.build:1:0: ERROR: Meson version is 0.64.1 but project requires >= 1.2.0

  A full log can be found at /build/contourpy-1.1.1/.mesonpy-6x3o8xj_/build/meson-logs/meson-log.txt
  + meson setup --prefix=/nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9 /build/contourpy-1.1.1 /build/contourpy-1.1.1/.mesonpy-6x3o8xj_/build --native-file=/b>
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9/bin/python3.10 /nix/store/9jhg0r6h6523a4fzjvsfzb3c1c64vfr8-python3.10-pip-22.3.1/lib/python3.10>
  cwd: /build/contourpy-1.1.1
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```
I already have
```
    contourpy = prev.contourpy.override {
      preferWheel = true;
    };
    meson-python = prev.meson-python.override {
      preferWheel = true;
    };
```
Is there a way to override the `meson` version used by `contourpy`?

**Comment by cpcloud** [comment]
This should be fixed by https://github.com/nix-community/poetry2nix/pull/1329. Can you give that a try?

**Comment by cpcloud** [comment]
Closed by #1329.
